src/wp-includes/post.php

```diff
@@ -8070,10 +8070,11 @@ function get_available_post_mime_types( $type = 'attachment' ) {
 	$mime_types = apply_filters( 'pre_get_available_post_mime_types', null, $type );
 
 	if ( ! is_array( $mime_types ) ) {
-		$mime_types = $wpdb->get_col( $wpdb->prepare( "SELECT DISTINCT post_mime_type FROM $wpdb->posts WHERE post_type = %s", $type ) );
+		$mime_types = $wpdb->get_col( $wpdb->prepare( "SELECT DISTINCT post_mime_type FROM $wpdb->posts WHERE post_type = %s AND post_mime_type != ''", $type ) );
 	}
 
-	return $mime_types;
+	// Remove nulls from returned $mime_types.
+	return array_filter( $mime_types );
```

Comment:
As `array_filter()` preserves the keys, this may result in gaps if the array was indexed.
I'm not sure if that's an issue, it's just that I'd expect a sequental keys returned by the `get_available_post_mime_types()` function.
Here's an example: https://3v4l.org/FIfCm

Reply:
It doesn't seem like that would be an issue in how the array is ultimately iterated. [In `post.php` it uses](https://github.com/WordPress/wordpress-develop/blob/204a1bbf4e5f22b07a93c1f4a0b12bdd65d6483f/src/wp-includes/post.php#L3377)
```php
foreach ( (array) $real_mime_types as $real ) {
    ...
```
where `$real_mime_types` is set from `$avail_post_mime_types`.

That said, what do you think about
```suggestion
	return array_values( array_filter( $mime_types ) );
```
to leave things tidy, especially if the keys would be expected to be sequential?

Reply:
Yes, it looks good. As far as I can tell, `$mime_types` is always an array at this step. So, it should be safe to call these 2 functions.